src/transport-managers/transport-managers.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { z } from 'zod';
+import { CreateTransportManagerSchema } from './types';
+
+@Injectable()
+export class TransportManagersService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async store(body: z.infer<typeof CreateTransportManagerSchema>) {
+    const { transportId, userId } = CreateTransportManagerSchema.parse(body);
+
+    let result = await this.prismaService.transport.findFirst({
+      where: {
+        id: transportId,
+      },
+      select: {
+        id: true,
+      },
+    });
+    if (!result) throw new Error('Transporte não encontrado.');
```

Comment:
Neste caso aqui você pode usar o os Exceptions Filters https://docs.nestjs.com/exception-filters#built-in-http-exceptions

Exemplo: throw new BadRequestException

Reply:
@rodrigooler corrigido!